### ISSUE TITLE
Test results are inaccessible to red-green color-blind, blind and CI

### ISSUE THREAD

**Issue by JohnAlbin** [avajs/ava#2919]
## TLDR

I'm using AVA 4.0.0-rc.1 and have a few tests marked as ".failing" since they are known test failures. When looking at the test output, I expect that I can tell which tests are known failures, but known-failing tests look exactly like passing tests in some situations.

## Details

AVA's default output doesn't clearly show which of the tests are known failures because it only distinguishes between test successes and test failures using color. **When you only use color to distinguish between different things in an interface, you are making those things inaccessible to some people.**

Here's some AVA test results on my terminal. You can see that both known failures and successes have the ✔ checkmark prepended and they only differ based on their color:

<img width="845" alt="ava-output-with-color" src="https://user-images.githubusercontent.com/33429/147272339-59ff15dd-1afe-4526-bb74-2fb7ab9b06ff.png">

This is how those same AVA test results appear on GitHub's CI, which is similar to how red-green color-blind would see it. There's no way to tell which tests are known failures:

<img width="862" alt="ava-output-on-ci" src="https://user-images.githubusercontent.com/33429/147272421-0e0c2fc2-c3de-4976-bba0-d3db116b064b.png">

We need to re-think how these test results are displayed so they are accessible to everyone, including:
* developers who are red-green colorblind
* developers using screen readers
* developers who don't have a color-enabled terminal
* developers reading a plain text log file (like in a CI)

## Possible solutions

As a reference, here's a screenshot of skipped, failing, passing and known failing tests:
<img width="557" alt="ava-all-test-types" src="https://user-images.githubusercontent.com/33429/147272547-e1d40ffd-3bc3-4f4b-bde7-25f285f7437a.png">

Maybe known failures should be in "skipped" yellow with a ✖ cross instead of the checkmark. Known failures are more like skipped tests than anything else.

Also, right now failing tests only have a red ✖ with normal-colored descriptive text. Can we make the descriptive text of failing tests all in red? And maybe bold too? The bold isn't necessary since failures are printed again at the end of the test result list, but it helps the failures visually stick out from the rest of the list.

As for people who are using screen readers, I think we might need to add text to describe what `-`, `✖`, `✔` mean. Especially if we use `✖` for known and unknown fails. Maybe this?:

```
  ✔ PASS: Example › should be an example test
  - SKIP: Example › should be an example test
  ✖ FAIL: Example › should be an example test
  ✖ KNOWN FAIL: Example › should be an example test (or maybe SKIP FAIL or SKIPPED FAIL)
```

Here's a mockup with color and bold added:
<img width="456" alt="ava-test-result-mockup" src="https://user-images.githubusercontent.com/33429/147275873-1f79abd2-067c-469a-9238-af73110d4ded.png">

**Comment by novemberborn** [comment]
Thanks for the feedback @JohnAlbin!

Counter-intuitively known failures count as a success — it's only when they stop failing that there's a problem. It being hard to filter the output because the symbol is reused isn't necessarily a display issue.

I think we may be able to make some tweaks with the current implementation, though to really make strides I think we need to tackle https://github.com/avajs/ava/issues/2500.

**Comment by JohnAlbin** [comment]
> Counter-intuitively known failures count as a success

Yep. It's one of the features I love about ava. Being able to document known bugs with tests makes it easer to fix them.

I actually noticed this issue because I have some known test failures in my project's integration tests with a dependency. I wanted to show the dependency project which tests were failing by showing them the GitHub Actions log results, BUT there's no color in the log files so it's impossible for anyone to read the the publicly-accessible log files and tell which tests are failing.

> It being hard to filter the output because the symbol is reused isn't necessarily a display issue.

If we "add text to describe what `-`, `✖`, `✔` mean", I'm fine with known tests still using `✔` icon because you can still tell that this red text is a known failure:
```
✔ KNOWN FAIL: Example › should be an example test (or maybe SKIP FAIL or SKIPPED FAIL)
```

Personally, I would still prefer the `✖`, but I'm fine with the `✔`. It's a _passing test_ because it is _failing_ and marked as `.failing`, so there's definitely ambiguity about what the "right" icon should be. :)
